app/scripts/modules/core/pipeline/config/validation/targetImpedance.validator.ts

```diff
@@ -0,0 +1,69 @@
+import {module} from 'angular';
+
+import {
+  PIPELINE_CONFIG_SERVICE,
+  PipelineConfigService
+} from '../services/pipelineConfig.service';
+import {
+  IStageOrTriggerTypeConfig, IStageOrTriggerValidator, IValidatorConfig,
+  PipelineConfigValidator, PIPELINE_CONFIG_VALIDATOR
+} from './pipelineConfig.validator';
+import {NAMING_SERVICE, NamingService} from 'core/naming/naming.service';
+import {IStage} from 'core/domain/IStage';
+import {IPipeline} from 'core/domain/IPipeline';
+
+export interface ITargetImpedanceValidationConfig extends IValidatorConfig {
+  stageTypes?: string[];
+  stageType?: string;
+  message: string;
+}
+
+export class TargetImpedanceValidator implements IStageOrTriggerValidator {
+
+  static get $inject() { return ['pipelineConfigService', 'namingService']; }
+
+  constructor(private pipelineConfigService: PipelineConfigService, private namingService: NamingService) {}
+
+  public validate(pipeline: IPipeline,
+                  stage: IStage,
+                  validator: ITargetImpedanceValidationConfig,
+                  config: IStageOrTriggerTypeConfig): string {
+
+    const stagesToTest: IStage[] = this.pipelineConfigService.getAllUpstreamDependencies(pipeline, stage),
+          regions: string[] = stage['regions'] || [];
+    let allRegionsFound = true;
+
+    regions.forEach((region) => {
+      let regionFound = false;
+      stagesToTest.forEach((toTest) => {
+        if (toTest.type === 'deploy' && toTest['clusters'] && toTest['clusters'].length) {
+          toTest['clusters'].forEach((cluster: any) => {
+            const clusterName: string = this.namingService.getClusterName(cluster.application, cluster.stack, cluster.freeFormDetails);
+            if (clusterName === stage['cluster'] && cluster.account === stage['credentials'] && cluster.availabilityZones && cluster.availabilityZones.hasOwnProperty(region)) {
+              regionFound = true;
+            }
+          });
+        } else if (toTest.type === 'cloneServerGroup' && toTest['targetCluster'] === stage['cluster'] && toTest['region'] === region) {
```

Comment:
the bracket notation is required on strongly typed objects in Typescript. Since `toTest` here implements the `IStage` interface, any dynamic fields (i.e. fields not defined in the interface) can only be referenced with the bracket notation.

In the interface definition, the presence of dynamic fields is declared via this type of syntax:
```
[k: string]: any;
```

So, short answer is: yes.